pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/participantfiles/ParticipantFilesUseCase.java

```diff
@@ -0,0 +1,51 @@
+package org.broadinstitute.dsm.model.participantfiles;
+
+import java.net.URL;
+import java.util.Optional;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.db.dto.ddp.instance.DDPInstanceDto;
+import org.broadinstitute.dsm.exception.DownloadException;
+import org.broadinstitute.dsm.model.elastic.ESFile;
+import org.broadinstitute.dsm.model.elastic.search.ElasticSearchParticipantDto;
+import org.broadinstitute.dsm.route.participantfiles.SignedUrlResponse;
+import org.broadinstitute.dsm.service.FileDownloadService;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+
+@Slf4j
+@AllArgsConstructor
+public class ParticipantFilesUseCase {
+
+    private static String CLEAN = "CLEAN";
+    private String bucketName;
+    private String blobName;
+    private String ddpParticipantId;
+    private String fileGuid;
+    FileDownloadService fileDownloadService;
+    Optional<DDPInstanceDto> ddpInstanceDto;
+
+
+    public boolean isFileClean() {
```

Comment:
Optional: we can use more java-8 like features to make code shorter but it's not necessary:

implementation suggestion, method `isFileClean`:
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
```
public boolean isFileClean() {
        return Optional.ofNullable(ddpInstanceDto.orElseThrow().getEsParticipantIndex())
                .map(esIndex -> ElasticSearchUtil.getParticipantESDataByParticipantId(esIndex, ddpParticipantId)
                        .orElseThrow(() -> new RuntimeException("Participant ES Data is not found for " + ddpParticipantId)))
                .map(esPtDto -> esPtDto.getFiles().stream().filter(this::filterFileByFileGuid).findAny()
                        .orElseThrow(() -> new RuntimeException("Could not match any files with " + fileGuid))
                        .isFileClean())
                .orElse(false);
    }

    private boolean filterFileByFileGuid(ESFile file) {
        return file.guid.equals(fileGuid);
    }
```
    
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -